docs/calendar-draft.md

```diff
@@ -0,0 +1,315 @@
+
+```javascript
+// Force the Gregorian calendar:
+Temporal.Date.from("2019-12-06").withCalendar("gregory").weekOfYear;
+
+// Use a calendar from another source:
+Temporal.Date.from("2019-12-06").withCalendar(Intl.defaultCalendar).weekOfYear;
+Temporal.Date.from("2019-12-06").withCalendar(request.calendar).weekOfYear;
+```
+
+The Gregorian calendar, ID "gregory" ([why not "gregorian"?](https://github.com/tc39/ecma402/issues/212)), could be used to indicate the full ISO calendar.  So, "iso" would mean the partial version, and "gregory" would mean the full version.  Alternatively, "iso8601", the identifier defined by CLDR as "Gregorian calendar using the ISO 8601 calendar week rules", could be the identifier for the full ISO calendar.
+
+## Temporal.Date API changes
+
+### New Temporal.Date instance methods
+
+The following methods involving ISO conversion would be added:
+
+```javascript
+Temporal.Date.prototype.toISO = function(): Temporal.Date {
+	const isoDate = this.calendar[Temporal.Calendar.toISO](this);
+	// assert: isoDate.calendar === Temporal.Calendar.iso
+	return isoDate;
+}
+
+// Temporal.Date.prototype.with does *not* modify the calendar. A new method
+// is added for that:
+
+Temporal.Date.prototype.withCalendar = function(newCalendar: Calendar) {
+	const isoDate = this.toISO();  // note: call intrinsic version
+	const otherDate = newCalendar[Temporal.Calendar.fromISO](isoDate);
+	// assert: otherDate.calendar === newCalendar
+	return otherDate;
+}
+```
+
+### ISO strings with calendar hint
+
+Serialization to/from strings will still be supported.  To convert to a string, the `Temporal.Calendar.id` field of the calendar will be appended in `[]` following the date.  For example, `2019-12-06[hebrew]` refers to 2019-12-06 projected into the Hebrew calendar.
+
+```javascript
+Temporal.Date.prototype.toString = function() {
+	let calendarKeyword, isoDate;
+	if (this.calendar === Temporal.Calendar.iso) {
+		calendarKeyword = null;
+		isoDate = this;
+	} else {
+		calendarKeyword = this.calendar[Temporal.Calendar.identifier];
+		isoDate = this.toISO();  // call intrinsic
+	}
+	// return an ISO string for isoDate with calendar in brackets:
+	// "2019-12-06[hebrew]"
+}
+```
+
+### New behavior of Temporal.Date.from
+
+`Temporal.Date.from` would take a new optional `options` argument, with a single field `idToCalendar` specifying a function to map from identifiers to Calendar objects.  If not present, the `Intl.Calendar` namespace will be searched.
+
+Example implementation:
+
+```javascript
+Temporal.Date.from = function(thing: string | object, options: object) {
+	if (typeof thing === "string") {
+		object = // components of string
+	}
+
+	const isoDate = // a date in the ISO calendar with fields from object
+
+	if (typeof object.calendar === "string") {
+		const calendar = options?.idToCalendar?.(object.calendar)
+			?? Temporal.Calendar.idToCalendar(id);  // call intrinsic
```

Comment:
We don't want this implicit escape hatch. If a lookup function is provided, it must be treated as 100% authoritative.

Reply:
Added this as a comment.

Reply:
Follow-up in #294